server/src/main/java/org/apache/druid/segment/indexing/granularity/UniformGranularitySpec.java

```diff
@@ -57,18 +60,17 @@ public UniformGranularitySpec(
     this.segmentGranularity = segmentGranularity == null ? DEFAULT_SEGMENT_GRANULARITY : segmentGranularity;
 
     if (inputIntervals != null) {
-      List<Interval> granularIntervals = new ArrayList<>();
-      for (Interval inputInterval : inputIntervals) {
-        Iterables.addAll(granularIntervals, this.segmentGranularity.getIterable(inputInterval));
-      }
-      this.inputIntervals = ImmutableList.copyOf(inputIntervals);
-      this.wrappedSpec = new ArbitraryGranularitySpec(queryGranularity, rollup, granularIntervals);
+      // sort them
```

Comment:
Contract indicates that inputIntervals must be left "as is".... will remove sorting in next commit

Reply:
done